mlflow/models/evaluation/default_evaluator.py

```diff
@@ -258,6 +259,38 @@ def _do_plot(**kwargs):
 }
 
 
+def _evaluate_custom_metric(custom_metric, eval_df, builtin_metrics):
+    result = custom_metric(eval_df, builtin_metrics)
+    if result is None:
+        _logger.warning(f"Custom metric function '{custom_metric.__name__}' returned None.")
+        return None, None
+
+    def __validate_metrics_dict(metrics):
+        return isinstance(metrics, dict) and all(
+            isinstance(metric_name, str) and isinstance(metric_val, (int, float, np.number))
+            for metric_name, metric_val in metrics.items()
+        )
+
+    def __validate_artifacts_dict(artifacts):
+        return isinstance(artifacts, dict) and all(
+            isinstance(artifacts_name, str) for artifacts_name in artifacts.keys()
+        )
+
+    if isinstance(result, dict) and __validate_metrics_dict(result):
+        return result, None
+
+    if (
+        isinstance(result, tuple)
+        and __validate_metrics_dict(result[0])
+        and __validate_artifacts_dict(result[1])
+    ):
+        return result
+
+    raise MlflowException(
+        f"Custom metric '{custom_metric.__name__}' did not return in an expected format."
```

Comment:
Can we link to the docs for expected input formats and / or provide a brief summary of the supported formats?

Reply:
^This existed but I asked @MarkYHZhang to take it out to avoid duplicating information that's hard to maintain (because this particular error isn't in an obvious place).
We will link to the docs once they're published

Reply:
@apurva-koti We should link to the docs in this PR, else initial users of the API won't see this helpful doc link in the error message. @MarkYHZhang can you include a link to the expected docs location that will exist once this PR is merged and docs are published? Happy to help.

Reply:
Honestly it's probably fine to restore your earlier message describing the expected types @MarkYHZhang and then just replace it with a docs link when they're published. That way there's never an information gap

Reply:
Can we include a link to the docs? We just need to link to `mlflow.evaluate()`. We already do this here: https://mlflow.org/docs/latest/models.html#model-evaluation.